src/Ext/Techno/Body.cpp

```diff
@@ -743,6 +743,16 @@ bool TechnoExt::IsHealthInThreshold(TechnoClass* pObject, double min, double max
 	return (hp > 0 ? hp > min : hp >= min) && hp <= max;
 }
 
+bool TechnoExt::IsVeterancyInThreshold(TechnoClass* pTechno, double min, double max)
+{
+	double veterancy = 0.0;
+
+	if (pTechno->GetTechnoType()->Trainable)
+		veterancy = pTechno->Veterancy.Veterancy;
+
+	return veterancy >= min && veterancy <= max;
```

Comment:
One of the limits should be inclusive, other exclusive. Otherwise imagine if you make one warhead effect that works on 0-1, another that works on 1-2. With your implementation it's impossible to create non-overlapping ranges of effect in a good fashion. See how `IsHealthInThreshold` handles this.

I think the lower bound should be inclusive, and the upper bound should be exclusive. The reasoning is that the game itself uses it like this. [0, 1) is Rookie, [1, 2) is Veteran, and [2, +inf) is Elite. This means


```suggestion
	return veterancy >= min && veterancy < max;
```

should be used.

However, the same issue as with `IsHealthInThreshold` arises when you can't then cover all ranges in a nice way, hence why we made 0 HP follow a slightly different set of rules. Accepting ideas on how to handle this similarly.